grails-app/views/_core_theme_includes.gsp

```diff
@@ -1,4 +1,6 @@
 <link rel="stylesheet" type="text/css" href="${resource(dir: 'js', file: 'ext-3.3.1/resources/css/ext-all-notheme.css')}"/>
 <link rel="stylesheet" type="text/css" href="${resource(dir: 'js', file: 'ext-3.3.1/resources/css/xBaseTheme.css')}"/>
 
-<link rel="shortcut icon" href="${resource(dir: 'images', file: 'favicon.ico')}" type="image/x-icon"/>
+<g:if test="${grailsApplication.config.portal.icon}" >
```

Comment:
Exactly, this is why you want `portalBranding`. Instead of handling the logic here in a template, you have a class.